From fe7acc535f0e211a751196fc80cb5576680c5c19 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Mon, 11 Jan 2016 14:31:24 +0200
Subject: [PATCH 0830/1429] fsl-dpaa2: mac: Fix crash on error path

If the fixed-phy cannot be correctly registered, unregister_netdev()
receives a non-NULL, yet invalid phydev. Force the phydev reference to
NULL to avoid a crash on the probe routine's error path.

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 7d6a958..e295f05 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -732,6 +732,8 @@ probe_fixed_link:
 		netdev->phydev = fixed_phy_register(PHY_POLL, &status, NULL);
 		if (!netdev->phydev || IS_ERR(netdev->phydev)) {
 			dev_err(dev, "error trying to register fixed PHY\n");
+			/* So we don't crash unregister_netdev() later on */
+			netdev->phydev = NULL;
 			err = -EFAULT;
 			goto err_no_phy;
 		}
-- 
1.7.5.4

